s3-supported-api.md

```diff
@@ -0,0 +1,119 @@
+# S3 api
+
+## Supported service api
+1.  Lists all your buckets
+    `aws s3 ls`
+
+## Supported operations/api on Bucket
+
+1.  GetBucket (List objects in the specified bucket)  
+    `aws s3 ls s3://<your_bucket>`  
+    `aws s3api list-objects --bucket <your_bucket>`  
+    `aws s3api list-objects-v2 --bucket <your_bucket>`  
+2.  Put bucket  
+    `aws s3 mb s3://<your_bucket>`  
+    `aws s3api create-bucket --bucket <your_bucket>`  
+3.  HEAD bucket  
+    `aws s3api head-bucket --bucket <your_bucket>`  
+4.  Delete bucket  
+    `aws s3 rb s3://<your_bucket>`  
+    `aws s3api delete-bucket --bucket <your_bucket>`  
+5.  Put Bucket ACL  
+    `aws s3api put-bucket-acl --bucket <your_bucket>`  
+6.  Get Bucket ACL  
+    `aws s3api get-bucket-acl --bucket <your_bucket>`  
+7.  Multipart upload
+    7.1    Create multipart upload in bucket  
+           `aws s3api create-multipart-upload --bucket <your_bucket> --key <key>`  
+    7.2    Upload part  
+           `aws s3api upload-part --bucket <your_bucket> --key <key> --part-number 1 --body <part>`  
+    7.3    Lists parts of a multipart upload  
+           `aws s3api list-parts --bucket <your_bucket> --key <key> --upload-id <id>`  
+    7.4    Abort/Complete multipart  
+           `aws s3api complete-multipart-upload --multipart-upload <mpustruct_file> --bucket <your_bucket> --key <key> --upload-id <id>`  
+           `aws s3api abort-multipart-upload --bucket <your_bucket> --key <key> --upload-id <id>`  
+    7.5    Lists in-progress multipart uploads in a bucket  
+           `aws s3api list-multipart-uploads --bucket <your_bucket>`  
+8.  Put Bucket policy  
+    `aws s3api put-bucket-policy --bucket <your_bucket> --policy <policy_json_file>`  
+9. Get Bucket policy  
+    `aws s3api get-bucket-policy --bucket <your_bucket>`  
+10. Delete bucket policy  
+    `aws s3api delete-bucket-policy --bucket <your_bucket>`  
+11. Put Bucket tagging  
+    `aws s3api put-bucket-tagging --bucket <your_bucket> --tagging <tagging_json_file>`  
+12. Get Bucket tagging  
+    `aws s3api get-bucket-tagging --bucket <your_bucket>`  
+13. Delete bucket tagging  
+    `aws s3api delete-bucket-tagging --bucket <your_bucket>`  
+14. Get bucket location  
+    `aws s3api get-bucket-location --bucket <your_bucket>`  
+
+
+## Supported operations/api on Object
+
+1.  Put Object  
+    `aws s3api put-object --bucket <your_bucket> --key <key> --body <file>`  
+2.  Get Object  
+    `aws s3api get-object --bucket <your_bucket> --key <key> <save_to_file>`  
```

Comment:
even get-object we can do through aws s3 cp

Reply:
ok.